### PR TITLE
Add table_aws_eks_node_group. Closes #132

### LINKED ISSUE #132: Add table_aws_eks_node_group
**Describe the new table**
A clear and concise description of the table and the resource.

**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by rajeshbal65**

### CODE REVIEW

File: aws/table_aws_eks_cluster.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+import (
+	"context"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+
+	"github.com/aws/aws-sdk-go/service/eks"
+)
+
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:12]:
```suggestion
//// TABLE DEFINITION
```

File: aws/table_aws_eks_cluster.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listEksClusters(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:134]:
```suggestion
```

File: aws/table_aws_eks_cluster.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getEksCluster(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getEksCluster")
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:162]:
```suggestion
```

File: aws/table_aws_eks_cluster.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+			},
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksCluster,
+			},
+			{
+				Name:        "created_at",
+				Description: "The Time when the cluster was created.",
```
Comment (khushboo9024) [inline aws/table_aws_eks_cluster.go:39]:
```suggestion
				Description: "The Unix epoch timestamp in seconds for when the cluster was created.",
```

File: aws/table_aws_eks_node_group.go [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+			},
+			{
+				Name:        "status",
+				Description: "The current status of the managed node group.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "capacity_type",
+				Description: "The current status of the cluster.",
```
Comment (khushboo9024) [inline aws/table_aws_eks_node_group.go:78]:
```suggestion
				Description: "The capacity type of your managed node group.",
```

File: aws/table_aws_eks_node_group.go [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+			{
+				Name:        "node_group_arn",
+				Description: "The Amazon Resource Name (ARN) of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+				Transform:   transform.FromField("nodegroupArn"),
+			},
+			{
+				Name:        "cluster_name",
+				Description: "The Time when the cluster was created.",
```
Comment (khushboo9024) [inline aws/table_aws_eks_node_group.go:42]:
```suggestion
				Description: "The name of the cluster that the managed node group resides in.",
```

File: aws/table_aws_eks_node_group.go [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+			},
+			{
+				Name:        "node_role",
+				Description: "The IAM role associated with your node group.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "labels",
+				Description: "If the node group wasn't deployed with a launch template, then this is the remote access configuration that is associated with the node group.",
```
Comment (khushboo9024) [inline aws/table_aws_eks_node_group.go:120]:
```suggestion
				Description: "The Kubernetes labels applied to the nodes in the node group.",
```

File: aws/table_aws_eks_node_group.go [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listEksNodeGroup(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_eks_node_group.go:185]:
```suggestion
```

File: aws/table_aws_eks_node_group.go [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+		return nil, err
+	}
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getEksNodeGroup(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getEksCluster")
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_eks_node_group.go:219]:
```suggestion
```

File: aws-test/tests/aws_rds_db_cluster_snapshot/variables.tf [hunk 10]
```diff
@@ -47,6 +47,12 @@ data "null_data_source" "resource" {
   }
 }
 
+resource "null_resource" "delay" {
+  provisioner "local-exec" {
+    command = "sleep 60"
+  }
+}
+
 resource "aws_vpc" "my_vpc" {
```
Comment (khushboo9024) [inline aws-test/tests/aws_rds_db_cluster_snapshot/variables.tf:56]: Please revert back this changes

File: aws/table_aws_eks_node_group.go [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+import (
+	"context"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+
+	"github.com/aws/aws-sdk-go/service/eks"
+)
+
```
Comment (khushboo9024) [inline aws/table_aws_eks_node_group.go:12]:
```suggestion
//// TABLE DEFINITION
```

File: aws/table_aws_eks_node_group.go [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+				Name:        "capacity_type",
+				Description: "The current status of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "scaling_config",
+				Description: "The scaling configuration details for the Auto Scaling group that is associated with node group.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksNodeGroup,
```
Comment (khushboo9024) [inline aws/table_aws_eks_node_group.go:86]: We should try to keep all the JSON fields at end of the table, please change accordingly

File: aws/table_aws_eks_node_group.go [hunk 13]
```diff
@@ -0,0 +1,257 @@
...
+				Description: "If a launch template was used to create the node group, then this is the launch template that was used.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags attached to the node group.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksNodeGroup,
+				Transform:   transform.FromField("tags"),
```
Comment (khushboo9024) [inline aws/table_aws_eks_node_group.go:153]: If we are getting the tags in TagsMap format the why we are adding tags_src column . And also please keep the `tags` column in ` Standard columns` section

File: aws/table_aws_eks_node_group.go [hunk 14]
```diff
@@ -0,0 +1,257 @@
...
+				Description: "The name associated with an Amazon EKS managed node group.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("NodegroupName"),
+			},
+			{
+				Name:        "node_group_arn",
+				Description: "The Amazon Resource Name (ARN) of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+				Transform:   transform.FromField("nodegroupArn"),
```
Comment (khushboo9024) [inline aws/table_aws_eks_node_group.go:38]: Is `nodegroupArn` the correct field name ? Please check if we are getting the correctvalues for both list and get calls.

**Review by khushboo9024 — COMMENTED** [review]
Please add the  proper integration test and  docs as well.

**Comment by cbruno10** [comment]
@khushboo9024 Can you please re-review when you get a chance? Thanks!

**Review by khushboo9024 — COMMENTED** [review]
@rajeshbal65  Please add the integration test if ready otherwise please close the PR, can be reopen once tests are ready.

**Pull request closed without merging**